UserDetailsClient/UserDetailsClient.Core/Features/LogOn/B2CAuthenticationService.cs

```diff
@@ -0,0 +1,140 @@
+﻿using Microsoft.Identity.Client;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace UserDetailsClient.Core.Features.LogOn
+{
+    public class B2CAuthenticationService : IAuthenticationService
+    {
+        public static IPublicClientApplication PCA = null;
+
+        public static object ParentActivityOrWindow { get; set; }
+
+        static B2CAuthenticationService()
+        {
+            // default redirectURI; each platform specific project will have to override it with its own
+            PCA = PublicClientApplicationBuilder.Create(B2CConstants.ClientID)
+                .WithB2CAuthority(B2CConstants.Authority)
+                .WithRedirectUri($"msal{B2CConstants.ClientID}://auth")
+                .Build();
+        }
+
+        public B2CAuthenticationService()
+        {
+        }
+
+        public async Task<string> AcquireToken()
+        {
+            IEnumerable<IAccount> accounts = await PCA.GetAccountsAsync();
+            AuthenticationResult ar = await PCA.AcquireTokenSilent(B2CConstants.Scopes, GetAccountByPolicy(accounts, B2CConstants.PolicySignUpSignIn))
+                .WithB2CAuthority(B2CConstants.Authority)
+               .ExecuteAsync();
+            string token = ar.AccessToken;
+            return token;
+        }
+
+        public async Task<UserContext> ResetPassword()
+        {
+            AuthenticationResult ar = await PCA.AcquireTokenInteractive(B2CConstants.Scopes)
+                .WithPrompt(Prompt.NoPrompt)
+                .WithAuthority(B2CConstants.AuthorityPasswordReset)
+                .WithParentActivityOrWindow(ParentActivityOrWindow)
+                .ExecuteAsync();
+
+            var userContext = UpdateUserInfo(ar);
+
+            return userContext;
+        }
+
+        public async Task<UserContext> EditProfile()
+        {
+            IEnumerable<IAccount> accounts = await PCA.GetAccountsAsync();
+            // KNOWN ISSUE:
+            // User will get prompted 
+            // to pick an IdP again.
```

Comment:
this can be removed. we now have Prompt.NoPrompt, so the comment was never updated. thanks.

Reply:
@markTi nit: did you remove this comment?

---
In reply to: [280848811](https://github.com/Azure-Samples/active-directory-b2c-xamarin-native/pull/85#discussion_r280848811) [](ancestors = 280848811)